invenio_rdm_records/ext.py

```diff
@@ -141,6 +141,6 @@ def init_resource(self, app):
 
         # PID provider
         self.pid_provider_resource = RDMPIDProviderResource(
-            service=None,   # PIDS-FIXME
+            service=self.records_service,
```

Comment:
Hmm Taking into account how it has been growing from parent to child class just adding functionality I thought it was better like this... However checking the `ext.py` registration I see this:
```
self.records_service = RDMRecordService(
            RDMRecordServiceConfig,
            files_service=FileService(RDMFileRecordServiceConfig),
            draft_files_service=FileService(RDMFileDraftServiceConfig),
        )
self.subjects_service = subject_record_type.service_cls(
            config=subject_record_type.service_config_cls,
        )
```

And I start to doubt 😅 What makes a service qualify to be a separate service (like subjects)? or to make it an attribute service of the main RecordsService (like files)? my gut tells me dependency of config/functions.

The potenatially separated `PIDsService` requires knowledge about the `record_cls` in some cases to get the record and add the specific pid.

Linking to your question below about not doing `drafts.commit()`. If we do that (not commit) then we need to **fully rely** on the frontend to have control and remember things (which might be difficult for API requests, i.e. they need to keep the json and know which value was returned). In addition, then when publishing the record the pid needs to come from the data (which in other cases, like access comes from the saved record). Moreover, if we want to merely "reserve" a PID no-matter of the record, then we should not have the `records/pid_value/drafts` in the endpoint...

TL;DR: I could separate, but it needs knowledge of some config and functions, so it will be a separated inheritance. Another option is make it an "attribute" service like the files one.

Reply:
IRL, keep as it is :) Let's retake this in the near future, might require refactor.